packages/babel-plugin-proposal-import-defer/src/index.ts

```diff
@@ -0,0 +1,121 @@
+import { declare } from "@babel/helper-plugin-utils";
+import type { types as t } from "@babel/core";
+import type { Scope } from "@babel/traverse";
+import { defineCommonJSHook } from "@babel/plugin-transform-modules-commonjs";
+
+import syntaxImportDefer from "@babel/plugin-syntax-import-defer";
+
+export default declare(api => {
+  api.assertVersion(7);
+  // We need the explicit type annotation otherwise when using t.assert* ts
+  // reports that 'Assertions require every name in the call target to be
+  // declared with an explicit type annotation'
+  const t: typeof api.types = api.types;
```

Comment:
"Fun" fact: `const a = b` and `const a: typeof b = b` are not equivalent in TS. https://github.com/microsoft/TypeScript/issues/36931